cirq-core/cirq/transformers/eject_z.py

```diff
@@ -111,31 +111,48 @@ def map_func(op: 'cirq.Operation', moment_index: int) -> 'cirq.OP_TREE':
             return op
 
         # Move Z gates into tracked qubit phases.
-        if isinstance(gate, ops.ZPowGate) and (
-            eject_parameterized or not protocols.is_parameterized(gate)
-        ):
-            qubit_phase[op.qubits[0]] += gate.exponent / 2
-            return []
-
-        # Try to move the tracked phases over the operation via protocols.phase_by(op)
-        phased_op = op
-        for i, p in enumerate([qubit_phase[q] for q in op.qubits]):
-            if not single_qubit_decompositions.is_negligible_turn(p, atol):
-                phased_op = protocols.phase_by(phased_op, -p, i, default=None)
-        if phased_op is None:
-            return [dump_tracked_phase(op.qubits), op]
+        if isinstance(gate, ops.ZPowGate):
+            if eject_parameterized or not protocols.is_parameterized(gate):
+                qubit_phase[op.qubits[0]] += gate.exponent / 2
+                return []
+            else:
+                return op
+
+        if isinstance(gate, ops.XPowGate):
+
+            return ops.PhasedXPowGate(
+                exponent=gate.exponent, phase_exponent=-qubit_phase[op.qubits[0]] * 2
+            )(op.qubits[0])
+        if isinstance(gate, ops.YPowGate):
+            return ops.PhasedXPowGate(
+                exponent=gate.exponent, phase_exponent=0.5 - qubit_phase[op.qubits[0]] * 2
+            )(op.qubits[0])
+        if isinstance(gate, ops.PhasedXPowGate):
+            return ops.PhasedXPowGate(
+                exponent=gate.exponent,
+                phase_exponent=gate.phase_exponent - qubit_phase[op.qubits[0]] * 2,
+                global_shift=gate.global_shift,
+            )(op.qubits[0])
```

Comment:
If we decide to do this change, let's move this to a helper method, which is responsible to do the phasing, within this module

Reply:
Isn't that basically what `map_func` is?  It's a helper function within `eject_z` that is responsible for doing the phasing.  Moving that outside `eject_z` would make things messier I think, because you would need to add in qubit_phase and other variables as additional arguments, and I don't think it would add anything.  I'm not sure restructuring would make things cleaner here.